src/common/fetchClient.ts

```diff
@@ -97,6 +97,9 @@ export default class FetchClient {
     function appendParams(key: string, value: any, parentKey?: string) {
       const fullKey = parentKey ? `${parentKey}[${key}]` : key;
 
+      // Skip undefined values
+      if (value === undefined) return url;
```

Comment:
What does it mean to return `url` here, i'm not sure if i follow? This `buildUrl` function is feeling a bit gnarly tbh.

Also out of curiosity, what is the behavior for `null` value currently?

Reply:
Agreed it's a bit complex. I actually misread the nested functions - I was trying to return early and thought that `appendParams` was returning url but it's actually `buildUrl`, so I can just return here. Will push an update for that.

`null` gets passed along in the `else` case of `appendParams`. I think we want to allow that so you can explicitly set null e.g. in the case where you want to clear a value.

Reply:
> I think we want to allow that so you can explicitly set null e.g. in the case where you want to clear a value.

Yeah i'd be curious to double check if `null` gets decoded to `nil` on the elixir side, which I think is the behavior you are describing.